### DISCUSSION TITLE
Rotating dag_processor and scheduler logs in Airflow 2.6.3

### DISCUSSION THREAD

**Discussion by shaurya-sood** [apache/airflow#34648] [General]
We are running Airflow 2.6.3 on Kubernetes, deployed via Helm 1.7.0, and we are facing issues with `kubernetes.ephemeral_storage.usage` when the limits (1GB) are being hit on our scheduler pod, leading to unwanted restarts. We are looking for a way to rotate the scheduler and dag_processor logs to address this issue.

**Here are our current logging configurations:**
- `dag_processor_log_target` is set to `stdout`.
- `remote_logging` is enabled.
- `delete_local_logs` is set to `True`.
- `logging_level = DEBUG`.
- We do not use any custom logging class.

Despite the above settings, 

- dag_processor logs are still written to both stdout and {AIRFLOW_HOME}/logs/dag_processor_manager/dag_processor_manager.log.
- Scheduler logs are also being written to stdout and {AIRFLOW_HOME}/logs/scheduler.

It seems like the provided configurations mainly work for task logs, and we haven't found a way to rotate dag_processor and scheduler logs.

**Queries:**

- How shall we approach rotating dag_processor and scheduler logs?
- Are there any recommended practices for managing scheduler and dag_processor logs?
- Would it be possible to enhance log rotation features to include these logs in upcoming Airflow releases?

Any advice or suggestions are appreciated.

**Comment by Taragolis** [comment]
> How shall we approach rotating dag_processor and scheduler logs?
> Are there any recommended practices for managing scheduler and dag_processor logs?

If you need to rotate logs than better look at [logrotate](https://linux.die.net/man/8/logrotate)

> Would it be possible to enhance log rotation features to include these logs in upcoming Airflow releases?

Difficult to say, I guess the chances is small, because first of all this mechanism should be added for someone and also it should be maintained. Right now only simple bash script exists into the official docker image: https://github.com/apache/airflow/blob/main/scripts/docker/clean-logs.sh

**Reply by shaurya-sood** [reply]
We're considering making the following adjustments for log management. 

1. Redirection of `dag_processor_manager_log_location` to /dev/null:
We’re considering redirecting to /dev/null using this [config setting](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-processor-manager-log-location) to save space (around 500MB) on the scheduler pod. Insights on the importance and implications of these logs would be appreciated.

2. Adjustment of Retention in [logGroomerSidecar](https://github.com/apache/airflow/blob/b316be1c551e3f1b5569642689f5097e0e0aabeb/chart/values.yaml#L875) which basically runs the [clean-logs.sh](https://github.com/apache/airflow/blob/main/scripts/docker/clean-logs.sh):
We're thinking of reducing retentionDays to 5, as we are already sending logs to stdout and ingesting them into Kibana, negating the need for local retention. Your thoughts on this would be valuable.

Thanks in advance for your feedback and suggestions.
